packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -188,6 +188,24 @@ const MetricForm: FC<MetricFormProps> = ({
   const [showAdvanced, setShowAdvanced] = useState(advanced);
   const [hideTags, setHideTags] = useState(!current?.tags?.length);
   const [sqlOpen, setSqlOpen] = useState(false);
+  const currentDatasource = current?.datasource
+    ? getDatasourceById(current?.datasource)
+    : null;
+  // Only set the default to true for new metrics with no sql or an edited or
+  // duplicated one where the sql matches the default.
+  const [allowAutomaticSqlReset, setAllowAutomaticSqlReset] = useState(
+    !current ||
+      !current?.sql ||
+      (currentDatasource &&
+        current.type &&
+        current.name &&
+        current?.sql ===
+          getInitialMetricQuery(
+            currentDatasource,
+            current.type,
+            current.name
+          )[1])
+  );
```

Comment:
Rewrite this to use a callback function with early returns and comments.  Very difficult to follow the logic right now.  You may also want to use `useMemo` instead of `useState`.

Reply:
I did refactor this code some to be clearer, though I didn't use a callback function.  This is just for setting the default value and I am a bit unclear how a callback would be helpful here.